packages/remix/src/utils/instrumentServer.ts

```diff
@@ -321,16 +322,18 @@ function wrapRequestHandler(origRequestHandler: RequestHandler, build: ServerBui
   const routes = createRoutes(build.routes);
   const pkg = loadModule<ReactRouterDomPkg>('react-router-dom');
   return async function (this: unknown, request: Request, loadContext?: unknown): Promise<Response> {
-    const url = new URL(request.url);
+    const local = domain.create();
+    return local.bind(async () => {
+      const url = new URL(request.url);
+      const transaction = startRequestHandlerTransaction(url, request.method, routes, pkg);
```

Comment:
We should probably check if tracing is enabled before starting the transaction. (Applies to express as well.)